core/storage/feedback/gae_models_test.py

```diff
@@ -49,6 +52,64 @@ def test_put_function(self):
         feedback_thread_model.put()
         self.assertNotEqual(feedback_thread_model.last_updated, last_updated)
 
+    def test_raise_exception_by_mocking_collision(self):
+        # Test create method.
+        with self.assertRaisesRegexp(
+            Exception, 'Feedback thread ID conflict on create.'):
+            # Swap dependent method get_by_id to simulate collision every time.
+            with self.swap(
+                feedback_models.GeneralFeedbackThreadModel, 'get_by_id',
+                types.MethodType(
+                    lambda x, y: True,
+                    feedback_models.GeneralFeedbackThreadModel)):
+                feedback_models.GeneralFeedbackThreadModel.create(
+                    'exploration.exp_id.thread_id')
+
+        # Test generate_new_thread_id method.
+        with self.assertRaisesRegexp(
+            Exception,
+            'New thread id generator is producing too many collisions.'):
+            # Swap dependent method get_by_id to simulate collision every time.
+            with self.swap(
+                feedback_models.GeneralFeedbackThreadModel, 'get_by_id',
+                types.MethodType(
+                    lambda x, y: True,
+                    feedback_models.GeneralFeedbackThreadModel)):
+                (
+                    feedback_models.GeneralFeedbackThreadModel
+                    .generate_new_thread_id('exploration', 'entity_id')
+                )
+
+
+class GeneralFeedbackMessageModelTests(test_utils.GenericTestBase):
+    """Tests for the GeneralFeedbackMessageModel class."""
+
+    def test_raise_exception_by_mocking_collision(self):
+        with self.assertRaisesRegexp(
+            Exception, 'Feedback message ID conflict on create.'):
+            # Swap dependent method get_by_id to simulate collision every time.
+            with self.swap(
+                feedback_models.GeneralFeedbackMessageModel, 'get_by_id',
+                types.MethodType(
+                    lambda x, y: True,
+                    feedback_models.GeneralFeedbackMessageModel)):
+                feedback_models.GeneralFeedbackMessageModel.create(
+                    'thread_id', 'message_id')
+
+    def test_get_all_messages(self):
+        thread_id = feedback_services.create_thread(
+            'exploration', '0', None, 'subject 1', 'text 1')
+
+        model = feedback_models.GeneralFeedbackMessageModel.get(
+            thread_id, 0)
+        self.assertEqual(model.entity_type, 'exploration')
+
+        all_msg = (
+            feedback_models.GeneralFeedbackMessageModel
+            .get_all_messages(1, None))
+        self.assertEqual(all_msg[0][0].text, 'text 1')
```

Comment:
Not enough validation here. Validate as much of all_msg as possible.

Also don't use abbreviations, call it all_messages.

Reply:
Done